tonic-types/src/richer_error/mod.rs

```diff
@@ -0,0 +1,500 @@
+use prost::{DecodeError, Message};
+use prost_types::Any;
+use tonic::{codegen::Bytes, metadata::MetadataMap, Code};
+
+mod error_details;
+mod std_messages;
+
+use super::pb;
+
+pub use error_details::{vec::ErrorDetail, ErrorDetails};
+pub use std_messages::{BadRequest, FieldViolation, RetryInfo};
+
+trait IntoAny {
+    fn into_any(self) -> Any;
+}
+
+trait FromAny {
+    fn from_any(any: Any) -> Result<Self, DecodeError>
+    where
+        Self: Sized;
+}
+
+fn gen_details_bytes(code: Code, message: &String, details: Vec<Any>) -> Bytes {
+    let status = pb::Status {
+        code: code as i32,
+        message: message.clone(),
+        details,
+    };
+
+    Bytes::from(status.encode_to_vec())
```

Comment:
Why encode to vec instead of using BytesMut + BufMut and regular `encode`? So that we avoid the vec round trip?

Reply:
Using `BytesMut` indeed makes much more sense, thanks! It seems to me that using `BufMut` is not necessary in this case but I may be missing something.